src/components/Foods.js

```diff
@@ -0,0 +1,122 @@
+import React from 'react';
+import foodSeed from '../foods.json'
+import FoodBox from './FoodBox'
+import AddFood from './AddFood'
+import TodayFood from './TodayFood'
+
+
+class Foods extends React.Component {
+    constructor(props){
+        super(props)
+        this.state ={
+            foodList: foodSeed,
+            filteredFoods: foodSeed,
+            formShow: false,
+            TodayFoods: []
+        }
+    }
+
+    formAppear = () => {
+        this.setState({
+            formShow: true
+        })
+    }
+
+    handleAdd = (e) => {
+        e.preventDefault()
+        let foodName        = e.currentTarget.foodName.value
+        let foodCalories    = e.currentTarget.foodCalories.value
+        let foodImage       = e.currentTarget.foodImage.value
+        let foodQuantity    = e.currentTarget.foodQuantity.value
+        let newFood = [
+            {name:      foodName, 
+            calories:   foodCalories,
+            image:      foodImage,
+            quantity:   foodQuantity   
+            }, ...this.state.foodList]
+
+        this.setState({
+          foodList:       newFood,
+          filteredFoods:  newFood,
+          formShow:       false
+        })
+    }
+
+    handleSearch = (e) => { 
+      let searchName = e.currentTarget.value
+      let cloneFoods = this.state.foodList.filter((food) => {
+        return food.name.startsWith(searchName.toUpperCase())
+      })
+      this.setState({
+        filteredFoods: cloneFoods
+    })
+    }
+
+    handleQuantity = (e) => {
+
+    }
+
+    handleToday = (e) =>{
+        let foodName        = e.currentTarget.food.name.value
```

Comment:
your target values are coming out as undefined, and that is the reason the code is breaking at this point. The reason is, for a button, the currentTarget will be the button (this is not a form button to submit info, it is a regular button). and that one doesn't hold the information of the food that should be added.
To go around this you could use state on the FodBox component to then pass the information of the food inside the function as a parameter :)